src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MagicNumberCheck.java

```diff
@@ -524,10 +524,14 @@ private boolean isInIgnoreList(DetailAST ast) {
      * @return {@code true} if {@code ast} is in the scope of field declaration
      */
     private static boolean isFieldDeclaration(DetailAST ast) {
-        DetailAST varDefAST = ast;
-        while (varDefAST != null
-                && varDefAST.getType() != TokenTypes.VARIABLE_DEF) {
-            varDefAST = varDefAST.getParent();
+        DetailAST varDefAST = null;
+        DetailAST node = ast;
+        while (node.getType() != TokenTypes.OBJBLOCK) {
+            if (node.getType() == TokenTypes.VARIABLE_DEF) {
```

Comment:
-> It will not make any issue with the code you give. will add in test in next push 

-> Please look at this 
```
kevin@inspiron-15-5510:~/Desktop/check_style$ cat Test2.java 
public class Test2 {
 
}
interface check {
    int a = 90;
}
kevin@inspiron-15-5510:~/Desktop/check_style$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="MagicNumber">
    </module>
  </module>
</module>kevin@inspiron-15-5510:~/Desktop/check_style$ java -jar /home/kevin/Downloads/checkstyle-10.9.2-all.jar -c config.xml Test2.java
Starting audit...
Audit done.
kevin@inspiron-15-5510:~/Desktop/check_style$ 
```
-> previous logic looks for variable def and the return statement will give true if `varDefAst.getParent().getParent()` is class then it is field but the loop will iterate till it don't find variable def here is the mistake it should not look till root. all the new variable def ast would be child of obj block. 
I think in point 2 it is a mistake and i realise when i see return statement doesn't contain any rule for interface. I have not check in detailed but according to the  code i provide you can see the ast of it is. I think it might be a new issue 
```
INTERFACE_DEF -> INTERFACE_DEF [4:0]
|--MODIFIERS -> MODIFIERS [4:0]
|--LITERAL_INTERFACE -> interface [4:0]
|--IDENT -> check [4:10]
`--OBJBLOCK -> OBJBLOCK [4:16]
    |--LCURLY -> { [4:16]
    |--VARIABLE_DEF -> VARIABLE_DEF [5:4]
    |   |--MODIFIERS -> MODIFIERS [5:4]
    |   |--TYPE -> TYPE [5:4]
    |   |   `--LITERAL_INT -> int [5:4]
    |   |--IDENT -> a [5:8]
    |   |--ASSIGN -> = [5:10]
    |   |   `--EXPR -> EXPR [5:12]
    |   |       `--NUM_INT -> 90 [5:12]
    |   `--SEMI -> ; [5:14]
    `--RCURLY -> } [6:0]
```   

-> using the return statement we are checking whether it is in the field or not if we add any condition in while loop it might leads us to pitest error ot coverage error

Reply:
Done
main logic has not been changed so regression not require